addons/blocked_locations/tests/test_location.py

```diff
@@ -3,99 +3,116 @@
+        )
 
     def test06_prepare_blocked_msg_non_blocked_location(self):
-        """ Prepare the message of a non blocked location.
-        """
+        """Prepare the message of a non blocked location."""
         msg = self.test_location_01._prepare_blocked_msg()
-        self.assertEqual(msg[0], 'Location %s is not blocked.' %
-                         self.test_location_01.name)
- 
+        self.assertEqual(
+            msg[0], 
+            "Location %s is not blocked." % self.test_location_01.name
+        )
+
     def test07_onchange_blocked_to_false(self):
-        """ Check that blocked reason becomes empty when the
-            location is unblocked
+        """Check that blocked reason becomes empty when the
+        location is unblocked
         """
+        self.test_location_01.u_blocked_reason = "Stock Damaged"
         self.test_location_01.u_blocked = True
-        self.test_location_01.u_blocked_reason = 'stock damaged'
         msg = self.test_location_01._prepare_blocked_msg()
-        self.assertEqual(msg[0],
-                         'Location %s is blocked (reason: stock damaged).' %
-                         self.test_location_01.name)
+        self.assertEqual(
+            msg[0], 
+            "Location %s is blocked (reason: %s)." % (self.test_location_01.name, self.test_location_01.u_blocked_reason)
+        )
 
         self.test_location_01.u_blocked = False
         # simulate the triggering of the onchange
         self.test_location_01.onchange_u_blocked()
-        self.assertEqual(self.test_location_01.u_blocked_reason,'')
+        self.assertEqual(self.test_location_01.u_blocked_reason, "")
 
     def test08_check_reserved_quants(self):
-        """ Blocking a location with reserved quants will raise an error.
-        """
-        self.create_quant(self.apple.id, self.test_location_01.id, 10,
-                          reserved_quantity=10)
+        """Blocking a location with reserved quants will raise an error."""
+        self.create_quant(self.apple.id, self.test_location_01.id, 10, reserved_quantity=10)
+
+        with self.assertRaises(ValidationError) as e:
+            self.test_location_01.u_blocked_reason = "Stock Damaged"
+            self.test_location_01.u_blocked = True
+
+        self.assertEqual(
+            e.exception.name, "Location cannot be blocked because it contains reserved stock."
+        )
+
+    def test09_check_blocked_reason(self):
+        """Blocking a location with no reason given will raise and error"""
 
         with self.assertRaises(ValidationError) as e:
             self.test_location_01.u_blocked = True
 
-        self.assertEqual(e.exception.name,
-                'Location cannot be blocked because it contains reserved stock.')
+        self.assertEqual(
+            e.exception.name,
+            "A reason for blocking the locations is required when attempting to block a location.",
+        )
```

Comment:
It's probably worth adding a test to explicitly check that when we don't supply a message that an exception is raised

Reply:
Hi @williamsajr isn't that what this test already does? :)

Reply:
Yep you're right, I get confused before my first cup of coffee in the morning